packages/medusa/src/models/cart.js

```diff
@@ -18,7 +19,7 @@ class CartModel extends BaseModel {
     shipping_address: { type: AddressSchema, default: {} },
     items: { type: [LineItemSchema], default: [] },
     region_id: { type: String, required: true },
-    discounts: { type: [String], default: [] },
+    discounts: { type: [DiscountModel], default: [] },
```

Comment:
Note that this is not a schema. It is a BaseModel class which means mongoose won't be able to parse the schema. Do `DiscountModel.schema` instead